core/src/main/java/org/elasticsearch/ingest/PipelineStore.java

```diff
@@ -235,8 +235,16 @@ public Pipeline get(String id) {
         return result;
     }
 
-    private Pipeline constructPipeline(String id, Map<String, Object> config) throws Exception {
-        return factory.create(id, config, processorFactoryRegistry);
+    private WritePipelineResponse constructPipelineResponse(String id, Map<String, Object> config) {
```

Comment:
maybe just return null if all is ok? then we just need to do a null check at line 158.